app/frontend/src/features/auth/useAuthStore.test.ts

```diff
@@ -0,0 +1,368 @@
+    getIsJailedMock.mockResolvedValue({ isJailed: true });
+    addDefaultUser();
+    const { result } = renderHook(() => useAuthStore());
+    await act(() => result.current.authActions.updateJailStatus());
+    expect(result.current.authState.jailed).toBe(true);
+    expect(result.current.authState.authenticated).toBe(true);
+  });
+  it("sets jailed to false for non-jailed user", async () => {
+    getIsJailedMock.mockResolvedValue({ isJailed: false });
+    addDefaultUser();
+    const { result } = renderHook(() => useAuthStore());
+    await act(() => result.current.authActions.updateJailStatus());
+    expect(result.current.authState.jailed).toBe(false);
+    expect(result.current.authState.authenticated).toBe(true);
+  });
+  it("throws if not logged in", async () => {
+    logoutMock.mockResolvedValue(new Empty());
+    addDefaultUser();
+    const { result } = renderHook(() => useAuthStore());
+    await act(async () => await result.current.authActions.logout());
+    expect.assertions(1);
+    try {
+      await act(() => result.current.authActions.updateJailStatus());
+    } catch (e) {
+      expect(e).toBeDefined();
+    }
+  });
+});
+
+describe("updateUserProfile action", () => {
+  it("updates the store with the latest user profile info", async () => {
+    addDefaultUser();
+    const {
+      aboutMe,
+      aboutPlace,
+      radius,
+      countriesLivedList,
+      countriesVisitedList,
+      gender,
+      name,
+      occupation,
+    } = defaultUser;
+    const newUserProfileData = {
+      // Unchanged data
+      name,
+      gender,
+      occupation,
+      aboutMe,
+      aboutPlace,
+      countriesLived: countriesLivedList,
+      // Changed data
+      city: "New York",
+      lat: 40.7306,
+      lng: -73.9352,
+      radius,
+      hostingStatus: 3,
+      languages: ["English", "Finnish", "Spanish"],
+      countriesVisited: [...countriesVisitedList, "United States"],
+    };
+    updateProfileMock.mockResolvedValue(new Empty());
+    getUserMock.mockResolvedValue({
+      ...newUserProfileData,
+      countriesLivedList,
+      countriesVisitedList: newUserProfileData.countriesVisited,
+    });
+    const { result } = renderHook(() => useAuthStore());
+
+    await act(() =>
+      result.current.profileActions.updateUserProfile(newUserProfileData)
+    );
+
+    expect(updateProfileMock).toHaveBeenCalledTimes(1);
+    expect(updateProfileMock).toHaveBeenCalledWith(newUserProfileData);
+    expect(getUserMock).toHaveBeenCalledTimes(1);
+    expect(getUserMock).toHaveBeenCalledWith("aapeli");
+
+    const state = result.current.authState;
+
+    // Things that have been updated are being reflected
+    expect(state.user).toMatchObject({
+      city: "New York",
+      hostingStatus: 3,
+      languages: ["English", "Finnish", "Spanish"],
+      countriesLivedList: ["Australia", "Finland", "Sweden", "United States"],
+      countriesVisitedList: ["Australia", "United States"],
+    });
```

Comment:
```suggestion
    expect(state.user).toMatchObject({
      city: "New York",
      hostingStatus: 3,
      languages: ["English", "Finnish", "Spanish"],
      countriesLivedList: ["Australia", "Finland", "Sweden", "United States"],
      countriesVisitedList: ["Australia", "United States"],
      lat: 40.7306,
      lng: -73.9352
    });
```
Should check the new location properties too